injection.js

```diff
@@ -0,0 +1,43 @@
+
+const customCSS = ".ytd-rich-grid-renderer, .ytd-item-section-renderer, .ytd-compact-video-renderer { opacity: 0; pointer-events: none; }";
+
+const allowedPaths = ["/", "/watch"];
+
+// add styling
+function addStyle() {
+	let styleElement = document.createElement("style");
+
+	styleElement.id = "injected-css";
+
+	styleElement.type = "text/css";
+
+	styleElement.innerHTML = customCSS;
+
+	document.head.appendChild(styleElement);
+}
+
+// remove styling
+function removeStyle() {
+	let styleElement = document.getElementById("injected-css");
+	
+	styleElement.parentNode.removeChild(styleElement);
+}
+
+// check if current url is allowed, and exectute accordingly
+function URLchecker() {

```

Comment:
Nitpick
The name of this function should probably be something like `updateStyle` or `checkAndUpdateStyle`.
I think it gives just that little more information.